### DISCUSSION TITLE
Review existing extensions, such as _RUFNAME

### DISCUSSION THREAD

**Discussion by hartenthaler** [fisharebest/gedcom-name#20] [General]
The GEDCOM-L group discussed the NAME tag at https://wiki.genealogy.net/GEDCOM/NAME-Tag (in German). This discussion was not finalized.

There is information about _RUFNAME at
- https://wiki.genealogy.net/GEDCOM/NAME-Tag#Vereinbarung_zu_Rufnamen
- https://wiki.genealogy.net/GEDCOM/NAME-Tag#Behandlung_von_Rufnamen
- https://de.wikipedia.org/wiki/Rufname_(Deutschland) (German Wikipedia)
_RUFNAME is finally defined in the GEDCOM-L addendum: https://genealogy.net/GEDCOM/

There are many more user-defined GEDCOM "name" tags (see https://wiki.genealogy.net/GEDCOM/_Nutzerdef-Tag):

- _ADPN | INDI.NAME | 0 |   | Adopted name |   | BKW6, PAF5
- _AKA | NAME |   |   | auch bekannt alsalso known as | als Ersatz für Programme, die keine Mehrfachnamen verarbeitenas replacement for programs not accepting multiple NAMEs | Stammbaumdrucker, GenPlus, FTB, PAF, AQ3, WebT
- _AKAN | INDI.NAME | 0 |   | Also known as |   | BKW6, PAF5
- _BIRN | INDI.NAME | 0 |   | Birth name |   | BKW6, PAF5
- _CALL | INDI.NAME | 0 |   | Called (name) |   | BKW6+
- _CENN | INDI.NAME | 0 |   | Census name |   | BKW6
- _CURN | INDI.NAME | 0 |   | Current name |   | BKW6, PAF5
- _FARN | INDI.NAME | 0 |   | Farm name |   | BKW6, PAF5
- _FKAN | INDI.NAME | 0 |   | Formal name |   | BKW6
- _FRKA | INDI.NAME | 0 |   | Formerly known as |   | BKW6+
- _GERN | INDI.NAME | 0 |   | German name, Rufname |   | BKW6, PAF5
- _HEBN | INDI.NAME | 0 |   | Hebrew name |   | BKW6
- _HEIRATNAME | INDI | 0 |   | HeiratsnameName after marriage |   | AF2000
- _HNM | INDI.NAME | 0 |   | Hebrew name |   | BKW6, WebT
- _INDG | INDI.NAME | 0 |   | Indigenous (name) |   | BKW6+, Familienbuch 6.0
- _INDN | INDI.NAME | 0 |   | Indian name |   | BKW6
- _LNCH | INDI.NAME | 0 |   | Legal name change |   | BKW6+
- _MARN | INDI.NAME | 0 |   | Married name |   | BKW6
- _MARNM | INDI.NAME | 0 |   | Used for married name |   | BKW6+, AQ3, PAF5, Stammbaumdrucker, FTB, WebT, Ahnenblatt
- _MARRNAME | FAM | 0 |   | Ehename eines der Partner (Vor- und Nachnamen) | Verwendung wie INDI.NAME, also mit SubTags GIVN und SURN, Bezug zu HUSB oder WIFE dadurch, dass direkt im Anschluss zu diesen Tags genannt | Leg
- _MARRNAMEHUSB |   |   |   | Ehename des Ehemannes |   | Familienbuch
- _MARRNAMEWIFE |   |   |   | Ehename der Ehefrau |   | Familienbuch
- _NAM | HUSB,WIFE |   |   | Ehenamen des Mannes oder der Frau |   | GES2000 V.2010
- _NAME | ADDR | + | ADDR A3 | Used for name in the address group, postal name |   | FO, PAF5, Leg, GenPlus, WebT
- _NAMM |   |   |   |   |   | GES-W-2010 Aug 2012, Stammbaumdrucker
- _NAMS | INDI | 0 |   | Namesake (godparent), Namensvetter |   | FTM201x, WebT
- _NAMW |   |   |   |   |   | GES-W-2010 Aug 2012, Stammbaumdrucker
- _OTHN | INDI.NAME | 0 |   | Other name |   | BKW6
- _RELN | INDI.NAME | 0 |   | Religious name |   | BKW6
- _RUFNAME | INDI.NAME. | + | NAME Ruf.B2 | Rufname (in offiziellen deutschen Dokumenten der Vorname der unterstichen wurde)common used given names, underlined in official German documents |   | Familienbande, Stammbaumdrucker, Familienbuch, GDO, GenPlus, GHome, AGES!, PCA, Bro-Keeper, Ahnenblatt, WebT
- _SHON | INDI.NAME | 0 |   | Short name |   | BKW6
- _SLDN | INDI.NAME | 0 |   | Soldier name |   | BKW6
- _SURN | INDI |   |   | Sippenname |   | GES2000 V.2010

There are some user-defined GEDCOM name tags that do not start with an _ (as they should):
- AKA | also known as | FO, FTM, Leg, PAF4
- NAMG | Naming | Gen
- NAMM | Name ??? | Reu
- NAMR | Religious Name | FTW5, Gen, Reu
- NAMS | Name Sake | FO, TMG, Reu
- NOTRY | Name des Notars - Ehevertrag, Testament | USM6
- RINI | Name der Person, die Info in Quelle gefunden hat | USM6
- UGNM | Rufname | USM6

**Comment by WGroleau** [comment]
I don't know whether anyone has created an extension for French "dit" names, e.g., Hercule Grandchamp dit Cornelier.  But that's something that deserves some kind of support.
